### PR TITLE
Upgrade Kafka client version to 0.10.0.1

### PR DESCRIPTION
**Pull request by 0xNacho**

### CODE REVIEW

File: src/main/java/com/treelogic/proteus/serialization/ProteusSerializer.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+				output.writeDouble(cast.getX());
+				output.writeDouble(cast.getY());
+				output.writeInt(cast.getVarName());
+				output.writeDouble(cast.getValue());
+			}
+		}
+
+		@SuppressWarnings("unused")
+		@Override
+		public Row read(Kryo kryo, Input input, Class<Row> clazz) {
```
Comment (VenturaDelMonte) [inline src/main/java/com/treelogic/proteus/serialization/ProteusSerializer.java:79]: Here, you should not instantiate according to Class<Row> variable but according to the type. You may find useful to check how i did the very same [thing](https://github.com/VenturaDelMonte/proteus-job/blob/f86bcd46013b1aab460aad7a506ea097879414a5/src/main/scala/eu/proteus/job/operations/serializer/CoilMeasurementKryoSerializer.scala) on the Flink side.

File: src/main/java/com/treelogic/proteus/serialization/ProteusSerializer.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import org.apache.kafka.common.serialization.Serializer;
+
+import java.io.Closeable;
+import java.util.Map;
+
+public class ProteusSerializer implements Closeable, AutoCloseable, Serializer<Row>, Deserializer<Row> {
+	private ThreadLocal<Kryo> kryos = new ThreadLocal<Kryo>() {
+		protected Kryo initialValue() {
+			Kryo kryo = new Kryo();
+			kryo.addDefaultSerializer(Row.class, new KryoInternalSerializer());
```
Comment (VenturaDelMonte) [inline src/main/java/com/treelogic/proteus/serialization/ProteusSerializer.java:22]: you should also register the subclasses here just to be safe.

File: src/main/java/com/treelogic/proteus/model/Row.java [hunk 3]
```diff
@@ -3,119 +3,113 @@
...
-        result = timestamp != null ? timestamp.hashCode() : 0;
-        result = 31 * result + coilId;
-        temp = Double.doubleToLongBits(x);
-        result = 31 * result + (int) (temp ^ (temp >>> 32));
-        result = 31 * result + (varName != null ? varName.hashCode() : 0);
-        temp = Double.doubleToLongBits(value);
-        result = 31 * result + (int) (temp ^ (temp >>> 32));
-        return result;
-    }
+	protected final int MAGIC_NUMBER = 0x00687691;
```
Comment (VenturaDelMonte) [inline src/main/java/com/treelogic/proteus/model/Row.java:8]: i think you may omit this here. it is needed only for the serialization part. However, it is not a big deal to have it here too.

**Review by VenturaDelMonte — COMMENTED** [review]
Overall this PR is well-made. There a few issues with the deserialization operation. @jeyhunkarimov will provide more insight on the kafka producer.

File: src/main/java/com/treelogic/proteus/serialization/ProteusSerializer.java (hunk 2 above)
Comment (0xNacho) [reply src/main/java/com/treelogic/proteus/serialization/ProteusSerializer.java:22]: Do you mean something like this? 

```java

private ThreadLocal<Kryo> kryos = new ThreadLocal<Kryo>() {
		protected Kryo initialValue() {
			Kryo kryo = new Kryo();
			KryoInternalSerializer kryoInternal = new KryoInternalSerializer();
			kryo.addDefaultSerializer(Row.class, kryoInternal);
			kryo.addDefaultSerializer(Row1D.class, kryoInternal);
			kryo.addDefaultSerializer(Row2D.class, kryoInternal);
			return kryo;
		};
	};

```

File: src/main/java/com/treelogic/proteus/model/Row.java (hunk 3 above)
Comment (0xNacho) [reply src/main/java/com/treelogic/proteus/model/Row.java:8]: :+1:

**Pull request closed without merging**